public/js/widgets/address-autocomplete.js

```diff
@@ -0,0 +1,146 @@
+let stylesInitialized = false;
+
+class AddressAutocomplete extends HTMLElement {
+	constructor()
+	{
+		super();
+		this.input = null;
+		this.wrapper = null;
+		this.row = null;
+		this.autocompleteChoices = null;
+		this.api = '';
+		this.getPredictionsRoute = Routing.generate('bgm_ajax_get_address_predictions', {});
+		this.getDetailsRoute = Routing.generate('bgm_ajax_get_address_details', {});
+
+		this.init();
+	}
+
+	init()
+	{
+		this.input = this.querySelector('input');
+		this.wrapper = this;
+		this.row = this.parentNode.parentNode;
+		this.autocompleteChoices = this.querySelector('.autocomplete-choices');
+		this.api = this.dataset.api;
+
+		this.querySelector('.loading-overlay').setAttribute('style', '');
+
+		this.initEvents();
+		this._injectCSS();
```

Comment:
Ces actions là devraient plutôt être callées dans le `connectedCallback()` du ton élément, au lieu du constructeur.

Explications de la différence entre les deux ici: https://stackoverflow.com/questions/40492330/difference-between-constructor-and-connectedcallback-in-custom-elements-v1

Donc tu pourrais surement juste renommer `init()` pour `connectedCallback()`, retirer le call de `this.init()` du constructeur, and you'd be good to go.

Reply:
Je suis encore noob avec les webcomponents mais c'est good to know ! J'ignorais que ça existait 😮 

Fixed it !